offline/cable_LUC_EXPT.F90

```diff
@@ -477,40 +370,16 @@ SUBROUTINE LUC_EXPT_INIT(LUC_EXPT)
        NotPrimOnly_fID = -1
     ENDIF
 
-    IF (TRIM(cable_user%MetType) .EQ. "bios") THEN
-       WHERE (LUC_EXPT%prim_only .eqv. .TRUE.)
-          LUC_EXPT%secdf    = 0.0
-          LUC_EXPT%primaryf = 1.0
-          LUC_EXPT%grass    = 0.0
-          LUC_EXPT%grass    = LUC_EXPT%primaryf * (1.0-CPC)
-          LUC_EXPT%primaryf = LUC_EXPT%primaryf * CPC
-       ENDWHERE
-       DEALLOCATE (CPC)
-    ELSE
-       ! set secondary vegetation area to be zero where land use transitions don't occur
-       ! set grass component of primary vegetation cover
-       WHERE (LUC_EXPT%prim_only .eqv. .TRUE.)
-          LUC_EXPT%secdf    = 0.0
-          LUC_EXPT%primaryf = 1.0
-          LUC_EXPT%grass    = 0.0
-          WHERE (LUC_EXPT%biome .eq. 3 .or. LUC_EXPT%biome .eq. 11) ! savanna/ xerophytic woods
-             LUC_EXPT%grass    = LUC_EXPT%primaryf * 0.6
-             LUC_EXPT%primaryf = LUC_EXPT%primaryf * 0.4
-          ELSEWHERE (LUC_EXPT%biome .eq. 12 .or. LUC_EXPT%biome .eq. 13 &
-               .or. LUC_EXPT%biome .eq. 15 .or. LUC_EXPT%biome .eq. 16  ) ! shrub
-             LUC_EXPT%grass    = LUC_EXPT%primaryf * 0.8
-             LUC_EXPT%primaryf = LUC_EXPT%primaryf * 0.2
-          ELSEWHERE (LUC_EXPT%biome .eq. 7 .or. LUC_EXPT%biome .eq. 8 &
-               .or. LUC_EXPT%biome .eq. 9 .or. LUC_EXPT%biome .eq. 10) ! boreal
-             LUC_EXPT%grass    = LUC_EXPT%primaryf * 0.2
-             LUC_EXPT%primaryf = LUC_EXPT%primaryf * 0.8
-          ELSEWHERE (LUC_EXPT%biome .eq. 5 .or. LUC_EXPT%biome .eq. 6 ) ! DBL
-             LUC_EXPT%grass    = LUC_EXPT%primaryf * 0.3
-             LUC_EXPT%primaryf = LUC_EXPT%primaryf * 0.7
-          END WHERE
-       END WHERE
-    ENDIF
-
+    ! set secondary vegetation area to be zero where land use transitions don't occur
```

Comment:
I suspect that I would reorganise this routine so that similar sections are co-located.  There's a lot of setting then resetting things as you go along which is somewhat confusing.

Reply:
It could most likely be improved. But at this point I am afraid of breaking things. Maybe something for later.